opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndexScan.java

```diff
@@ -62,37 +66,41 @@ public class OpenSearchIndexScan extends TableScanOperator {
    */
   public OpenSearchIndexScan(OpenSearchClient client,
                              Settings settings, String indexName,
-                             OpenSearchExprValueFactory exprValueFactory) {
-    this(client, settings, new OpenSearchRequest.IndexName(indexName), exprValueFactory);
+                             PlanContext context, OpenSearchExprValueFactory exprValueFactory) {
+    this(client, settings, new OpenSearchRequest.IndexName(indexName), context, exprValueFactory);
   }
 
   /**
    * Constructor.
    */
   public OpenSearchIndexScan(OpenSearchClient client,
-      Settings settings, OpenSearchRequest.IndexName indexName,
-      OpenSearchExprValueFactory exprValueFactory) {
+                             Settings settings, OpenSearchRequest.IndexName indexName,
+                             PlanContext context, OpenSearchExprValueFactory exprValueFactory) {
     this.client = client;
-    this.request = new OpenSearchQueryRequest(indexName,
-        settings.getSettingValue(Settings.Key.QUERY_SIZE_LIMIT), exprValueFactory);
+    this.maxResultWindow = context.getMaxResultWindow();
+    switch (context.getIndexScanType()) {
```

Comment:
i not sure how the PlanContext help in this case. seems we could build OpenSerachRequest when visit the logical plan tree.